examples/ensemble/plot_stack_predictors.py

```diff
@@ -60,35 +144,79 @@ def plot_regression_results(ax, y_true, y_pred, title, scores, elapsed_time):
 #
 # Here, we combine 3 learners (linear and non-linear) and use a ridge regressor
 # to combine their outputs together.
+#
+# Note: although we will make new pipelines with the processors which we wrote
+# in the previous section for the 3 learners, the final estimator RidgeCV()
+# does not need preprocessing of the data as it will be fed with the already
+# preprocessed output from the 3 learners.
 
-from sklearn.ensemble import StackingRegressor
-from sklearn.ensemble import RandomForestRegressor
-from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+
+from sklearn.experimental import enable_hist_gradient_boosting
 from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.ensemble import StackingRegressor
 from sklearn.linear_model import LassoCV
 from sklearn.linear_model import RidgeCV
 
-estimators = [
-    ('Random Forest', RandomForestRegressor(random_state=42)),
-    ('Lasso', LassoCV()),
-    ('Gradient Boosting', HistGradientBoostingRegressor(random_state=0))
-]
-stacking_regressor = StackingRegressor(
-    estimators=estimators, final_estimator=RidgeCV()
-)
 
+lasso_pipe = make_pipeline(processor_lin,
+                           LassoCV())
+
+rf_pipe = make_pipeline(processor_nlin,
+                        RandomForestRegressor(random_state=42))
+
+gradient_pipe = make_pipeline(processor_nlin,
+                              HistGradientBoostingRegressor(random_state=0))
+
+estimators = [('Random Forest', rf_pipe),
+              ('Lasso', lasso_pipe),
+              ('Gradient Boosting', gradient_pipe)]
+
+stacking_regressor = StackingRegressor(estimators=estimators,
+                                       final_estimator=RidgeCV())
 
+
+###############################################################################
+# Measure and plot the results
 ###############################################################################
-# We used the Boston data set (prediction of house prices). We check the
-# performance of each individual predictor as well as the stack of the
+#
+# Now we can use Ames Housing dataset to make the predictions. We check the
+# performance of each individual predictor as well as of the stack of the
 # regressors.
+#
+# The function ``plot_regression_results`` is used to plot the predicted and
+# true targets.
+
 
 import time
+import matplotlib.pyplot as plt
 import numpy as np
-from sklearn.datasets import load_boston
 from sklearn.model_selection import cross_validate, cross_val_predict
 
-X, y = load_boston(return_X_y=True)
+
+def plot_regression_results(ax, y_true, y_pred, title, scores, elapsed_time):
+    """Scatter plot of the predicted vs true targets."""
+    ax.plot([y_true.min(), y_true.max()],
+            [y_true.min(), y_true.max()],
+            '--r', linewidth=2)
+    ax.scatter(y_true, y_pred, alpha=0.2)
```

Comment:
I find it more intuitive to put `y_pred` on the x axis. What do you and other people think?

Alternatively we could do a residual plot:

```
ax.scatter(y_pred, y_true - y_pred, alpha=0.2)
```

which would make it possible to highlight that the Gaussian homo-schedastic noise assumption of the linear model does not hold for this data. It might hold better if you wrap the Lasso model with a TransformedTargetRegressor with `func=np.log` and `inverse_func=np.exp`. But then reporting R2 score on such transformed data is not very meaning ful.

Maybe we should just transform the price once and for all when loading the data and cast the problem as a log-price regression problem at the beginning. That might make the example simpler and the linear model should behave better and maybe contribute positively to the `StackingRegressor` ensemble (instead of negatively at the moment).

Reply:
Once the [MAPE PR](https://github.com/scikit-learn/scikit-learn/issues/10708) is reviewed and merged we could update this example to compute MAPE on the price in $ (by taking the `np.exp` of the predictions and `y_test` if `y_test` has been log-transformed when loading the data).